### PR TITLE
Soft audit fixes mbare

### PR DESCRIPTION
**Pull request by mbare0**

### CODE REVIEW

File: contracts/Pool.sol [hunk 1]
```diff
@@ -38,9 +38,10 @@ contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC20 {
       CompoundingIndex depositIndex_,
       CompoundingIndex borrowIndex_
     )  initializer public {
+      if(!_checkIfContract(address(borrowersRegistry_))) revert MustBeContract();
```
Comment (kavsky02) [inline contracts/Pool.sol:41]: Don't we need the same check for the borrowersRegistry?

File: contracts/Pool.sol [hunk 2]
```diff
@@ -324,6 +327,17 @@ contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC20 {
   }
 
 
+  function _checkIfContract(address account) internal returns(bool) {
```
Comment (kavsky02) [inline contracts/Pool.sol:330]: This can be private I think?

File: src/components/HistoryList.vue [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 
 <script>
-  import { transactionUrl } from "../utils/blockchain";
+  import { transactionUrl } from "../utils/sync-fork";
```
Comment (kavsky02) [inline src/components/HistoryList.vue:17]: Is this frontend change in scope of this PR?

File: contracts/Pool.sol (hunk 2 above)
Comment (mbare0) [reply contracts/Pool.sol:330]: right

File: src/components/HistoryList.vue (hunk 3 above)
Comment (mbare0) [reply src/components/HistoryList.vue]: this line is wrong, reverting

File: contracts/Pool.sol (hunk 1 above)
Comment (mbare0) [reply contracts/Pool.sol:41]: do you mean RatesCalculator? The fact that it's a contract is gonna be indirectly checked when calling `_updateRates()`

Comment (kavsky02) [reply contracts/Pool.sol:41]: That's what I meant. Okay this makes sense, thanks.

**Review by kavsky02 — APPROVED** [review]
LGTM

File: contracts/Pool.sol [hunk 4]
```diff
@@ -324,6 +327,17 @@ contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC20 {
   }
 
 
+  function _checkIfContract(address account) private returns(bool) {
```
Comment (jakub-wojciechowski) [inline contracts/Pool.sol:330]: Could we simply use: 
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Address.sol

We've discussed it and I don't see is as a vulnerability but to avoid getting silly comments from auditors let's also switch to the "send" method from this library instead of transfer:
https://docs.openzeppelin.com/contracts/2.x/api/utils#Address-sendValue-address-payable-uint256-

Comment (jakub-wojciechowski) [reply contracts/Pool.sol:330]: BTW, the second change could be done as a separate issue, not to block this PR from merging

Comment (mbare0) [reply contracts/Pool.sol]: does the second comment about `send` refer to all `transfer` methods used in contracts?

Comment (mbare0) [reply contracts/Pool.sol]: repaced with OpenZeppelin Address, replacing `send` in a separate issue

**Review by kamilovsky01 — APPROVED** [review]
LGTM

**Review by c-gma — APPROVED** [review]
